product-service/services/index.ts

```diff
@@ -0,0 +1,52 @@
+import { PRODUCTS_TABLE_NAME, STOCKS_TABLE_NAME } from "../constants";
+import { performQuery, performQueryWithTransaction } from "../db";
+import { Product } from "../models/db";
+
+const productFields = ['id', 'title', 'description', 'price', 'count'];
+
+
+export const getProducts = async (): Promise<Product[]> => {
+  const query = `SELECT ${productFields} FROM ${PRODUCTS_TABLE_NAME} inner join ${STOCKS_TABLE_NAME} on id = product_id where count > 0;`;
+  return performQuery<Product>(query);
+}
+
+export const getProductById = async (id: string): Promise<Product> => {
+  const query = {
+    text: `SELECT ${productFields} FROM ${PRODUCTS_TABLE_NAME} inner join ${STOCKS_TABLE_NAME} on id = product_id where id::TEXT = $1;`,
+    values: [id]
+  }
+  const [product] = await performQuery<Product>(query);
+  if (!product) {
+    throw new Error(`Product with id = [${id}] was not found`);
+  }
+  return product;
+}
+
+export const createProduct = async (productToCreate: Product) => {
+  const productQuery = {
+    text: `INSERT INTO
+        ${PRODUCTS_TABLE_NAME}(title, description, price)
+        VALUES($1, $2, $3) RETURNING *`,
```

Comment:
Maybe it is better to return just `id`, because only `id` we will need to cretae stock

Reply:
thanks for your review, fixed